swap/swap.go

```diff
@@ -466,6 +569,8 @@ func (s *Swap) SentCheques() (map[enode.ID]*Cheque, error) {
 // ReceivedCheque returns the last received cheque for a given peer
 func (s *Swap) ReceivedCheque(peer enode.ID) (cheque *Cheque, err error) {
 	if swapPeer := s.getPeer(peer); swapPeer != nil {
+		swapPeer.lock.Lock()
```

Comment:
Not an expert in go concurrency but is a lock (especially a write lock) really needed when accessing a getter for a primitive type value?

Reply:
you're right, this makes no sense. 

fixed